### PR TITLE
ARROW-6086: [Rust] [DataFusion] Add support for partitioned Parquet data sources

### PR DESCRIPTION
**Pull request by andygrove**
I discovered this last minute while running manual tests. I have been able to run parallel queries against parquet files using this branch as a dependency.

### CODE REVIEW

**Comment by andygrove** [comment]
Sorry @paddyhoran I found this last minute issue ... pretty small fix though.

File: rust/datafusion/src/datasource/parquet.rs [hunk 1]
```diff
@@ -37,23 +37,27 @@ use parquet::file::reader::*;
...
-            schema,
-        })
+    pub fn try_new(path: &str) -> Result<Self> {
+        let mut filenames: Vec<String> = vec![];
+        common::build_file_list(path, &mut filenames, ".parquet")?;
+        if filenames.is_empty() {
+            Err(ExecutionError::General("No files found".to_string()))
+        } else {
+            let parquet_file = ParquetFile::open(&filenames[0], None, 0)?;
+            let schema = parquet_file.projection_schema.clone();
```
Comment (paddyhoran) [inline rust/datafusion/src/datasource/parquet.rs]: What happens if the `schema` of the files differ?  I guess it just fails are execution time when a different `schema` is encountered?

**Review by paddyhoran — APPROVED** [review]
LGTM, just one question.

File: rust/datafusion/src/datasource/parquet.rs (hunk 1 above)
Comment (andygrove) [reply rust/datafusion/src/datasource/parquet.rs]: Yes, this code assumes that all of the partitions have the same schema currently. It's pretty basic. I imagine we could eventually have schema merging.

Comment (andygrove) [reply rust/datafusion/src/datasource/parquet.rs]: I will write up a JIRA to add validation that all the partitions have the same schema. That would be a nice improvement.

Comment (paddyhoran) [reply rust/datafusion/src/datasource/parquet.rs]: Makes sense.  Feel free to merge.

**Pull request closed without merging**